### PR TITLE
Precache pubmed data before starting dump

### PR DESCRIPTION
**Pull request by aparton**
As ENSVAR-2114

### CODE REVIEW

File: modules/Bio/EnsEMBL/VEP/Pipeline/DumpVEP/InitDump.pm [hunk 1]
```diff
@@ -115,6 +124,8 @@ sub generate_species_jobs {
   if($var_db_name) {
     my $has_sift_poly = $self->has_sift_poly($dbc, $var_db_name, $species_id);
     $species_hash{$_} = $has_sift_poly->{$_} for keys %$has_sift_poly;
+    
+    my $pubmed = $self->pubmed($dbc_var, $current_db_name, $species_id);
```
Comment (ens-lgil) [inline modules/Bio/EnsEMBL/VEP/Pipeline/DumpVEP/InitDump.pm]: I am not sure if this is relevant to have the ref hash returned as it is not used in the method.

File: modules/Bio/EnsEMBL/VEP/Pipeline/DumpVEP/InitDump.pm [hunk 2]
```diff
@@ -128,6 +139,76 @@ sub generate_species_jobs {
...
+      unlink($lock) if -e $lock;
+      my $sleep_count = 0;
+      if(-e $lock) {
+        while(-e $lock) {
+          sleep 1;
+          die("I've been waiting for $lock to be removed for $sleep_count seconds, something may have gone wrong\n") if ++$sleep_count > 900;
+        }
+      }
+      if(-e $file) {
+        open IN, $file;
```
Comment (ens-lgil) [inline modules/Bio/EnsEMBL/VEP/Pipeline/DumpVEP/InitDump.pm]: Do we actually need to open and read the file here ?

File: modules/Bio/EnsEMBL/VEP/Pipeline/DumpVEP/Dumper/Variation.pm [hunk 3]
```diff
@@ -426,7 +421,7 @@ sub pubmed {
         die("I've been waiting for $lock to be removed for $sleep_count seconds, something may have gone wrong\n") if ++$sleep_count > 900;
```
Comment (ens-lgil) [inline modules/Bio/EnsEMBL/VEP/Pipeline/DumpVEP/Dumper/Variation.pm]: Is the code related to the `lock` file still relevant here ?
The `InitDump.pm` step is supposed to be finished before this `Dumper/Variation` step starts.

We might need to display a warning if the `$file` doesn't exist (in the following lines).

**Comment by at7** [comment]
merged into master and release/98

**Pull request merged**